### PR TITLE
improve canberra metric tests

### LINKED ISSUE #1765: Improve `ignite.contrib.metrics.regression` tests
We have improved the tests for most of the metrics in `ignite.contrib.metrics` module, now we need to updated the tests for the regression metrics and add distributed tests, just like we have done with these: [AveragePrecision](https://github.com/pytorch/ignite/blob/master/tests/ignite/contrib/metrics/test_average_precision.py), [CohenKappa](https://github.com/pytorch/ignite/blob/master/tests/ignite/contrib/metrics/test_cohen_kappa.py).
I suggest making several PRs, each PR contains improvements for a one metric, this way we will be able to solve the issues if we got an issues related to a specific test, and more than one person can work on this issue (pick a metric + work on it)

### PR DESCRIPTION
**Pull request by KickItLikeShika**
Fixes #1765 

Added integration tests.

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/contrib/metrics/regression/test_canberra_metric.py [hunk 1]
```diff
@@ -63,6 +64,42 @@ def test_compute():
     assert canberra.pairwise([v1, v2])[0][1] == pytest.approx(np_sum)
 
 
+def test_integration_binary_input_with_output_transform():
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/regression/test_canberra_metric.py:67]: This kind of granularity "_with_output_transform" looks strange. Why do we test only with `output_transform` and not without ?

File: tests/ignite/contrib/metrics/regression/test_canberra_metric.py [hunk 2]
```diff
@@ -63,6 +64,42 @@ def test_compute():
...
+
+    test_cases = [
+        (torch.rand(size=(100,)), torch.rand(size=(100,)), 10),
+        (torch.rand(size=(200,)), torch.rand(size=(200,)), 10),
+        (torch.rand(size=(100,)), torch.rand(size=(100,)), 20),
+        (torch.rand(size=(200,)), torch.rand(size=(200,)), 20),
+    ]
+
+    for y_pred, y, batch_size in test_cases:
+        # check multiple random inputs as random exact occurencies are rare
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/regression/test_canberra_metric.py:98]: This statement is incorrect now. Data is already generated and we run 10 times on the same data.

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the PR!
I have a couple of comments here.

File: tests/ignite/contrib/metrics/regression/test_canberra_metric.py (hunk 1 above)
Comment (KickItLikeShika) [reply tests/ignite/contrib/metrics/regression/test_canberra_metric.py:67]: I always keep "with_output_transform" separated from "without_output_transform", i thought `test_compute` does testing "without_output_transform" part, i can add "without_output_transform" tests if needed..

File: tests/ignite/contrib/metrics/regression/test_canberra_metric.py (hunk 2 above)
Comment (KickItLikeShika) [reply tests/ignite/contrib/metrics/regression/test_canberra_metric.py:98]: I will add the `get_test_cases` part here

**Review by vfdev-5 — APPROVED** [review]
LGTM! Thanks @KickItLikeShika

**Pull request merged**